client/typescript/fine-uploader.d.ts

```diff
@@ -2984,397 +2315,1240 @@ declare namespace FineUploader {
+             * @param number id : A file id to apply these upload success parameters to
+             */
+            setUploadSuccessParams(newParams: any, id?: number): void;
+
+        }
+
+
+        /**
+         * AzureChunkingOptions
+         */
+        export interface AzureChunkingOptions extends ChunkingOptions {
+            /**
+             * The maximum size of each part, in bytes
+             *
+             * @default `5242880`
+             */
+            partSize: number;
+            /**
+             * Files smaller than this value will not be chunked.
+             *
+             * @default `4000001`
+             */
+            minFileSize?: number
+        }
+
+        /**
+         * AzureCorsOptions
+         */
+        export interface AzureCorsOptions extends CorsOptions {
+            /**
+             * Enables or disables cross-domain ajax calls (if the `expected` property is true) in IE9 and older.
+             *
+             * @default `true`
+             */
+            allowXdr: boolean;
+        }
+
+        /**
+         * AzureBlobPropertyNameFunction
+         */
+        export interface AzureBlobPropertyNameFunction {
+            constructor(id: number);
+        }
+
+        /**
+         * AzureBlobPropertyOptions
+         */
+        export interface AzureBlobPropertyOptions {
+            /**
+             * Describes the blob name used to identify the file in your Azure Blob Storage container.
+             *
+             * Possible values are
+             * * `'uuid'`
+             * * `'filename'`
+             * * `function`
+             *
+             * If the value is a function, Fine Uploader Azure will pass the associated file ID as a parameter when invoking your function.
+             * If the value is a function it may return one of a `qq.Promise` or a `String`
+             *
+             * @default `'uuid'`
+             */
+            name?: string | AzureBlobPropertyNameFunction;
+        }
+
+        /**
+         * AzureRequestOptions
+         */
+        export interface AzureRequestOptions extends RequestOptions {
+            /**
+             * URL for your Azure Blob Storage container
+             *
+             * @default `null`
+             */
+            containerUrl?: string;
+            /**
+             * Parameters passed along with each upload request.
+             *
+             * @default `{}`
+             */
+            params?: any;
+            /**
+             * Part of the parameter name that contains the name of the associated file which may differ from the blob name.
+             * Prefixed with 'x-ms-meta-' by Fine Uploader
+             *
+             * @default `'qqfilename'`
+             */
+            filenameParam?: string;
+        }
+
+        /**
+         * type for Azure's customHeaders function
+         */
+        export interface AzureCustomHeaderFunction {
+            constructor(id: number);
```

Comment:
@jclangst is there any reason to change this interface's declaration?
Before it was `(id: number): void;` and now it is `constructor(id: number);`